innodb/src/main/java/org/apache/calcite/adapter/innodb/InnodbRules.java

```diff
@@ -184,10 +184,13 @@ RelNode convert(LogicalFilter filter, InnodbTableScan scan) {
       final IndexCondition indexCondition =
           translator.translateMatch(filter.getCondition());
 
+      RexNode condition =
+          RexUtil.composeConjunction(cluster.getRexBuilder(),
+              indexCondition.getPushDownConditions());
```

Comment:
In InnodbFilter, condition contains both indexed and non-indexed conditions, but the actual execution only executes the indexed condition, which causes some rules to incorrectly optimize